db/migrate/20210713092440_add_permissions.rb

```diff
@@ -0,0 +1,24 @@
+class AddPermissions < ActiveRecord::Migration[6.0]
+  PRODUCT_PERMISSION_NAMES = [
+    'view_scc_products',
+    'subscribe_scc_products'
+  ].freeze
+  PERMISSION_NAMES_ALL = [
+    'test_connection_scc_accounts',
+    'view_scc_products',
+    'subscribe_scc_products'
+  ].freeze
+
+  def up
+    Permission.create!(:name => 'test_connection_scc_accounts', :resource_type => 'SccAccount')
+    PRODUCT_PERMISSION_NAMES.each do |p|
+      Permission.create!(:name => p, :resource_type => SccProduct)
```

Comment:
You could avoid having two arrays with almost the same content by doing something like the following:
```suggestion
  def up
    permissions = PERMISSION_NAMES_ALL.clone
    Permission.create!(:name => permissions.pop, :resource_type => 'SccAccount')
    permissions.each do |p|
      Permission.create!(:name => p, :resource_type => SccProduct)
```

Reply:
I agree that this is much better code, but is it more readable?

Reply:
I agree that messing with a copy of the array is not that nice.

Alternatively, I'd settle for always using `PRODUCT_PERMISSION_NAMES` and doing the destroy of `test_connection_scc_accounts` separately or adding it to the array locally:
```ruby
  def down
    (PRODUCT_PERMISSION_NAMES + ['test_connection_scc_accounts']).each do |p|
      Permission.where(:name => p).destroy_all
    end
  end

Reply:
I like that one!